tests/scripts/local_monitoring_tests/rest_apis/test_list_metadata.py

```diff
@@ -0,0 +1,475 @@
+    true or false
+    """
+
+    print("\n****************************************************")
+    print("Test verbose = ", verbose)
+    print("****************************************************\n")
+
+    form_kruize_url(cluster_type)
+    input_json_file = "../json_files/import_metadata.json"
+
+    response = delete_metadata(input_json_file)
+    print("delete metadata = ", response.status_code)
+
+    # Import metadata using the specified json
+    response = import_metadata(input_json_file)
+    metadata_json = response.json()
+
+    # Validate the json against the json schema
+    errorMsg = validate_import_metadata_json(metadata_json, import_metadata_json_schema)
+    assert errorMsg == ""
+
+    json_data = json.load(open(input_json_file))
+    datasource = json_data['datasource_name']
+    # Currently only default cluster is supported by Kruize
+    cluster_name = "default"
+
+    response = list_metadata(datasource=datasource, cluster_name=cluster_name, verbose=verbose)
+
+    list_metadata_json = response.json()
+    assert response.status_code == SUCCESS_200_STATUS_CODE
+
+    if verbose == "true":
+        # Validate the json against the json schema
+        errorMsg = validate_list_metadata_json(list_metadata_json, list_metadata_json_verbose_true_schema)
+        assert errorMsg == ""
+    else:
+        # Validate the json against the json schema
+        errorMsg = validate_list_metadata_json(list_metadata_json, list_metadata_json_cluster_name_without_verbose_schema)
+        assert errorMsg == ""
+
+    # Validate the json values
+    import_metadata_json = read_json_data_from_file(input_json_file)
+    validate_list_metadata_parameters(import_metadata_json, list_metadata_json, cluster_name=cluster_name)
+
+    response = delete_metadata(input_json_file)
+    print("delete metadata = ", response.status_code)
+
+
+@pytest.mark.sanity
+def test_list_metadata_datasource_cluster_name_and_namespace(cluster_type):
+    """
+    Test Description: This test validates GET /dsmetadata by passing a valid datasource, cluster name, namespace and
+    Note: Verbose is set to be true by default when namespace parameter is passed
+    """
+
+    form_kruize_url(cluster_type)
+    input_json_file = "../json_files/import_metadata.json"
+
+    response = delete_metadata(input_json_file)
+    print("delete metadata = ", response.status_code)
+
+    # Import metadata using the specified json
+    response = import_metadata(input_json_file)
+    metadata_json = response.json()
+
+    # Validate the json against the json schema
+    errorMsg = validate_import_metadata_json(metadata_json, import_metadata_json_schema)
+    assert errorMsg == ""
+
+    json_data = json.load(open(input_json_file))
+    datasource = json_data['datasource_name']
+    # Currently only default cluster is supported by Kruize
+    cluster_name = "default"
+    namespace = "monitoring"
```

Comment:
Tested the updated code on both minikube & openshift 
[PR1199_local_monitoring_tests_minikube.zip](https://github.com/user-attachments/files/15510653/PR1199_local_monitoring_tests_minikube.zip)
[PR1199_local_monitoring_tests_openshift.zip](https://github.com/user-attachments/files/15510657/PR1199_local_monitoring_tests_openshift.zip)

Reply:
In mandatory parameters missing test, there is a change required in the error message

```
Mandatory parameters missing [dataSourceName]  - this should be datasource_name
```

Reply:
delete metadata returns 405 instead of 200, can you please check this

```
Deleting the metadata...
URL =  http://192.168.49.2:31830/dsmetadata
<Response [405]>
Response status code =  405
delete metadata =  405
```

Reply:
In the test, rest_apis/test_import_metadata.py::test_repeated_metadata_import, introduce some changes by creating new namespaces and deleting one of the namespace and validate if the second import output reflects these changes.

Reply:
I see 4 negative test failures, are you yet to fix these?

Reply:
> In mandatory parameters missing test, there is a change required in the error message
> 
> > Mandatory parameters missing [dataSourceName]  - this should be datasource_name

`dataSourceName` variable is referred from the [DSMetadataAPIObject field](https://github.com/kruize/autotune/blob/mvp_demo/src/main/java/com/autotune/analyzer/serviceObjects/DSMetadataAPIObject.java#L25) during validation

Reply:
`delete metadata - 405` and current test failures - `test_list_metadata_after_deleting_metadata` & `test_list_metadata_without_import_action` will be resolved once PR 1178 is merged

Reply:
> In the test, rest_apis/test_import_metadata.py::test_repeated_metadata_import, introduce some changes by creating new namespaces and deleting one of the namespace and validate if the second import output reflects these changes.

Are these changes made?

Reply:
Updated test `rest_apis/test_import_metadata.py::test_repeated_metadata_import` by  dynamically creating and deleting namespaces to showcase the delete metadata API functionality

test log output [test_repeated_metadata_import.txt](https://github.com/user-attachments/files/15530854/test_repeated_metadata_import.txt)